### PR TITLE
Workshops/list of workshops

### PR DESCRIPTION
**Pull request by mkeerthiasm**
added- GET/Workshop_id
Modified -POST/Workshop_create
Modified - GET/workshop/ returns list of workshops

### CODE REVIEW

**Review by SiddonsD — APPROVED** [review]

File: exception_al/exception_al/settings.py [hunk 1]
```diff
@@ -35,6 +35,8 @@
     'rest_framework',
     'rest_framework.authtoken',
     'django.contrib.admin',
+    'workshops.apps.WorkshopsConfig', # This is the WorkshopsConfig class from the apps.py file
+    'rest_framework', # This is the Django REST framework
```
Comment (suanTech) [inline exception_al/exception_al/settings.py:39]: Please remove this line

File: exception_al/workshops/models.py [hunk 2]
```diff
@@ -18,7 +19,7 @@ class Workshop(models.Model):
     support_target = models.PositiveIntegerField()
     est_cost = models.DecimalField(max_digits=10, decimal_places=2)
     materials = models.TextField(blank=True)
-    status = models.CharField(max_length=255)
+    is_open = models.BooleanField(default=True)
     created_by = models.ForeignKey(User, on_delete=models.CASCADE)
```
Comment (suanTech) [inline exception_al/workshops/models.py:23]: Please replace User with get_user_model() from django.contrib.auth

File: exception_al/workshops/views.py [hunk 3]
```diff
@@ -1,20 +1,71 @@
...
+    
+class WorkshopListView(APIView):
 
-    def post(self, request, format=None):
+    def get(self, request):
+        workshops = Workshop.objects.all()
+        serializer = WorkshopSerializer(workshops, many=True)
+        return Response(serializer.data)
+    
+    def post(self, request):
```
Comment (suanTech) [inline exception_al/workshops/views.py:27]: Please add a permission classes
permission_classes = [permissions.IsAuthenticatedOrReadOnly]

imported from rest_framework

File: exception_al/workshops/views.py [hunk 4]
```diff
@@ -1,20 +1,71 @@
...
         serializer = WorkshopSerializer(data=request.data)
         if serializer.is_valid():
             serializer.save()
             return Response(serializer.data, status=status.HTTP_201_CREATED)
         return Response(serializer.errors, status=status.HTTP_400_BAD_REQUEST)
+    
+# This is the view for getting a single workshop by id    
+
+class WorkshopDetailView(APIView):
+
```
Comment (suanTech) [inline exception_al/workshops/views.py:37]: Please add 
   permission_classes = [IsOwnerOrReadOnly]

We'll need to create a custom permission class in a new permissions.py file like this
```
from rest_framework import permissions

class IsOwnerOrReadOnly(permissions.BasePermission):
  def has_object_permission(self, request, view, obj):
    if request.method in permissions.SAFE_METHODS:
      return True
    return obj.owner == request.user
```

**Review by suanTech — CHANGES_REQUESTED** [review]
Please fix the commented lines and push the changes.

File: exception_al/workshops/views.py (hunk 3 above)
Comment (mkeerthiasm) [reply exception_al/workshops/views.py:27]: Added IsAuthenticatedOrReadOnly

File: exception_al/exception_al/settings.py (hunk 1 above)
Comment (mkeerthiasm) [reply exception_al/exception_al/settings.py:39]: Done

File: exception_al/workshops/models.py (hunk 2 above)
Comment (mkeerthiasm) [reply exception_al/workshops/models.py:23]: Done replaced get_user_model()

File: exception_al/workshops/views.py (hunk 4 above)
Comment (mkeerthiasm) [reply exception_al/workshops/views.py:37]: Done! created permissions class accordingly.

**Pull request closed without merging**